assets/prototype/domain/shared/predicates/datetimes/filters/soldOutOnly/index.test.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Internal dependencies
+ */
+import soldOutOnly from './index';
+
+describe('soldOutOnly', () => {
+	it('Should return empty array if dates are not active, nor upcoming', () => {
+		const filteredDates = soldOutOnly([
+			{ id: 'abc', isDeleted: true },
+			{ id: 'def', isSoldOut: false },
+			{ id: 'def', status: 'ACTIVE' },
+		]);
+
+		expect(filteredDates).toEqual([]);
+	});
+
+	it('Should return an array of soldOutOnly date when isSoldOut is true and have a valid status', () => {
+		const filteredDates = soldOutOnly([
+			{ id: 'abc', isActive: false },
+			{ id: 'def', isSoldOut: true, status: 'ACTIVE' },
+			{ id: 'xyz', isSoldOut: true, status: 'WHATEVER' },
+		]);
+
+		expect(filteredDates.length).toBe(2);
+		expect(filteredDates[0].id).toBe('def');
+		expect(filteredDates[1].id).toBe('xyz');
+	});
+
+	it('Should return an array of soldOutOnly date when isSoldOut is true and have a valid status', () => {
+		const filteredDates = soldOutOnly([
+			{ id: 'abc', capacity: 2, sold: 2 },
+			{ id: 'def', capacity: 20, sold: 10 },
+			{ id: 'xyz', capacity: 100, sold: 100 },
+		]);
```

Comment:
test name is duplicate of preceding test and does not reflect this one, so should likely change.
maybe something like:

```
it('Should return an array of soldOutOnly date when capacity is equal to amount sold', () => {
```

Reply:
Thx :bowing_man: 
Had updated here: https://github.com/eventespresso/event-espresso-core/pull/2174/commits/d93c75fa4c8238c2802775a4125cb2b19fbb10d1